pkg/controller/tidb_control.go

```diff
@@ -81,8 +92,18 @@ func (tdc *defaultTiDBControl) ResignDDLOwner(tc *v1alpha1.TidbCluster, ordinal
 	tcName := tc.GetName()
 	ns := tc.GetNamespace()
 
+	scheme := "http"
+	if tc.Spec.EnableTLSServer {
+		scheme = "https"
+		rootCAs, _ := httputil.ReadCACerts()
```

Comment:
Are the certs the same for different components? If so, then this can be done once when tidb-controller-manager starts and use when needed.